src/containers/BooksList.js

```diff
@@ -18,30 +19,49 @@ class BooksList extends React.Component {
 
   render() {
     const { books } = this.props;
-    const BooksList = books.books.map(
-      b => (
-        <Book
-          key={b.id}
-          Id={b.id}
-          title={b.title}
-          cat={b.category}
-          Click={() => this.handleRemove(b.id)}
-        />
-      ),
-    );
+    let BooksList;
+    if (books.filter === 'All') {
+      BooksList = books.books.map(
+        b => (
+          <Book
+            key={b.id}
+            Id={b.id}
+            title={b.title}
+            cat={b.category}
+            Click={() => this.handleRemove(b.id)}
+          />
+        ),
+      );
+    } else {
+      const booksFiltered = books.books.filter(book => book.category === books.filter);
+      BooksList = booksFiltered.map(
+        b => (
+          <Book
+            key={b.id}
+            Id={b.id}
+            title={b.title}
+            cat={b.category}
+            Click={() => this.handleRemove(b.id)}
+          />
+        ),
+      );
+    }
 
     return (
-      <Table striped bordered hover>
-        <thead>
-          <tr>
-            <th>BookID</th>
-            <th>Title</th>
-            <th>Category</th>
-            <th>Delete</th>
-          </tr>
-        </thead>
-        {BooksList}
-      </Table>
+      <div>
+        <CategoryFilter />
+        <Table striped bordered hover>
```

Comment:
`handleFilterChange` should be passed to `<CategoryFilter/>` as `prop`
[see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png)]

Reply:
Perfect,